src/main/java/pl/edu/pw/elka/paprykaisalami/geeruh/issues/domain/ports/IssueService.java

```diff
@@ -114,4 +114,25 @@ public Either<DomainError, Issue> assignUser(IssueId issueId, @Nullable UserId u
                     return issueRepository.save(issue);
                 });
     }
+
+    @Transactional
+    @Valid
+    public Either<DomainError, Issue> relateIssue(IssueId issueId, IssueId relatedIssueId) {
+        return issueRepository.findById(issueId).map(
+                issue -> {
+                    issue.getRelatedIssues().add(relatedIssueId);
```

Comment:
You should avoid nested calls, especially when it comes to domain objects. It would better to make a function like `relateTo()` / `breakRelationWith()` or something like that.